project/categories/routes.py

```diff
@@ -0,0 +1,175 @@
+# Modules from Flask Library
+from flask import Flask, render_template, request, flash, Blueprint
+from flask import session as login_session
+from flask import make_response, redirect, jsonify, url_for
+# SQLAlchemy library to access database
+from sqlalchemy import create_engine, asc
+from sqlalchemy.orm import sessionmaker
+from project.models import Base, Category, Product, User
+
+import random
+import string
+import httplib2
+import json
+import requests
+
+# Authentification modules
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.client import FlowExchangeError
+
+# Create Blueprint 'categories'
+categories = Blueprint('categories', __name__)
+
+# Connect to Database and create database session
+engine = create_engine('sqlite:///furniturecatalog.db',
+                       connect_args={'check_same_thread': False})
+# Bind the engine to the metadata of the Base class so that the
+# declaratives can be accessed through a DBSession instance
+Base.metadata.bind = engine
+
+DBSession = sessionmaker(bind=engine)
+# A DBSession() instance establishes all conversations with the database
+# and represents a "staging zone" for all the objects loaded into the
+# database session object. Any change made against the objects in the
+# session won't be persisted into the database until you call
+# session.commit(). If you're not happy about the changes, you can
+# revert all of them back to the last commit by calling
+# session.rollback()
+session = DBSession()
+
+
+@categories.route('/')
+@categories.route('/categories/')
+def showCategories():
+    """App route function for main page to show all categories."""
+    # Check if user is logged in
+    isLogin = 'email' in login_session
+    # All categories ordered by name
+    categories = session.query(Category).order_by(Category.name).all()
+    # Select recently added products
+    latestProducts = (session.query(Product)
+                      .order_by(Product.created_date.desc()).limit(8))
+    return render_template('index.html', categories=categories,
+                           isLogin=isLogin, latestProducts=latestProducts)
```

Comment:
This is a very good start!
The next step would be to decouple logic from the routes and create services. For example, you could create `services` subdirectory and `categories.py` in it. This file would contain `CategoryService` with a set of methods, which you could call from your routes. For ex:

### services/categories.py:
```python
from project.db import DBConnector # this does not exist, you have to create it
from project.models import Category

class CategoryService():

    def __init__(self):
        self.session = DBConnector().get_session()

    def get_all_categories(self):
        return self.session.query(Category).order_by(Category.name).all()

    def get_category_by_id(self, category_id):
        return self.session.query(Category).filter_by(id=cat_id).one()
```
### categories/routes.py
Once you do this for categories, products, and authentication, `showCategories` could look like this:
```python
import AuthService, CategoryService, ProductService

auth_service = AuthService()
category_service = CategoryService()
product_service = ProductService()

@categories.route('/')
@categories.route('/categories/')
def showCategories():
    """App route function for main page to show all categories."""
    return render_template(
        'index.html',
        categories=category_service.get_all_categories(),
        isLogin=auth_service.is_user_legged_in(),
        latestProducts=product_service.get_latest_products(limit=8)
    )
```